### PR TITLE
Enable thermald for Dell XPS 15 7590

### PR DESCRIPTION
**Pull request by ghisvail**
###### Description of changes

Enable thermald.

###### Things done

- [x] Tested the changes in your own NixOS Configuration
- [x] Tested the changes end-to-end by using your fork of `nixos-hardware` and
      importing it via `<nixos-hardware>` or Flake input

### CODE REVIEW

**Comment by ghisvail** [comment]
I also noticed that previous and next XPS models have ` boot.kernelParams = lib.mkDefault [ "acpi_rev_override" ];` but not this one. I wonder whether it's intentional or a mistake.

**Comment by Mic92** [comment]
bors merge

**Comment by Mic92** [comment]
Don't know if we need the acpi flag here or not: 
https://cateee.net/lkddb/web-lkddb/ACPI_REV_OVERRIDE_POSSIBLE.html

**Comment by ghisvail** [comment]
Another thing is that the `nixpkg.overlay` section to override the WiFI module does not seem to be required anymore. I was able to configure my laptop just fine without it.

Should I submit a PR to remove it?

**Comment by Mic92** [comment]
> Another thing is that the `nixpkg.overlay` section to override the WiFI module does not seem to be required anymore. I was able to configure my laptop just fine without it.
> 
> Should I submit a PR to remove it?

sure.

File: dell/xps/15-7590/default.nix [hunk 1]
```diff
@@ -21,6 +21,9 @@
   # Prevent small EFI partiion from filling up
   boot.loader.grub.configurationLimit = 10;
 
+  # This will save you money and possibly your life!
+  services.thermald.enable = lib.mkDefault true;
```
Comment (Mic92) [inline dell/xps/15-7590/default.nix:25]: `lib` needs to be added to the header of this module:

```
+ { lib, ...}:
```

Comment (ghisvail) [reply dell/xps/15-7590/default.nix:25]: my bad, missed it when copy pasting from my config

**Comment by Mic92** [comment]
bors merge

**Pull request merged**